pkg/controller/bootstrap/bootstrapsigner.go

```diff
@@ -307,3 +296,21 @@ func (e *Signer) getTokens() map[string]string {
 
 	return ret
 }
+
+// handleErr checks if an error happened and makes sure we will retry later.
+func (e *Signer) handleErr(err error, key interface{}) {
+	if err == nil {
+		e.syncQueue.Forget(key)
+		return
+	}
+
+	// This controller retries 5 times if something goes wrong. After that, it stops trying.
+	if e.syncQueue.NumRequeues(key) < 5 {
+		e.syncQueue.AddRateLimited(key)
+		return
+	}
```

Comment:
IIUC, the requests can fire one after another without any polling / delay between them.
5 seems arbitrary and i don't think it will help.